### ISSUE TITLE
Make the docs for the `server_capabilities table` more verbose.

### ISSUE THREAD

**Issue by Decipher-CS** [neovim/neovim#22535]
### Problem

`help vim.lsp.client` has a portion about `server_capabilities` which needs to be more descriptive. All you take away from the current description is that it's a table that's sent by the server. But it should ideally also describe what values the table has, such as `documentFormattingProvider` and `renameProvider`.

So either have  (1) a dedicated page to all the available options or (2) a link to whichever place provides and maintains all the information about it, i.e. [lsp server protocol specifications](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/)

### Expected behavior

Information on `{server_capabilities}` should be explicitly stated with either a dedicated section or a redirect to whoever has it.

**Comment by clason** [comment]
Full documentation is out of scope, but a (sufficiently stable) link to the specification sounds like a good idea. PR welcome!

**Comment by Decipher-CS** [comment]
> Full documentation is out of scope, but a (sufficiently stable) link to the specification sounds like a good idea. PR welcome!

I'll create a PR. You reckon this link from [Microsoft](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/) is good enough?

**Comment by clason** [comment]
Yes, if it's a link directly to a section that has the missing information. Finding something on this page is pretty much impossible otherwise.

(And be aware that much of the documentation is autogenerated, so you might have to find the corresponding docstring in the `runtime/lua/vim/lsp` files.)

**Comment by justinmk** [comment]
here's a more precise link: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#messageDocumentation

it mentions : 

> LSP defines a set of requests, responses and notifications. Each of those are documented using the following format:
> ...
> an optional Server Capability section describing the server capability of the request.

**Comment by Decipher-CS** [comment]
> here's a more precise link: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#messageDocumentation

Noted. 
Just a heads up, I am not familiar with the auto-docs thing and LSP in general so I'll need to read-up more on all this jazz before I can open a PR.
